pkg/controller/util/clustercidrconfig/controller_utils.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustercidrconfig
+
+import (
+	"fmt"
+
+	networkingv1alpha1 "k8s.io/api/networking/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/slice"
+)
+
+// CreateClusterCIDRConfigHandler creates clusterCIDRConfig handler.
+func CreateClusterCIDRConfigHandler(f func(ccc *networkingv1alpha1.ClusterCIDRConfig) error) func(obj interface{}) {
+	return func(originalObj interface{}) {
+		ccc := originalObj.(*networkingv1alpha1.ClusterCIDRConfig).DeepCopy()
```

Comment:
why is this needed? do we mutate objects on the handlers?

Reply:
No we don't mutate the objects on the handlers currently. Removing the DeepCopy, we can add later if ever in the future we have any mutable fields for CCC